src/main/java/org/sqlite/jdbc3/JDBC3PreparedStatement.java

```diff
@@ -353,16 +353,7 @@ public void setObject(int pos, Object value) throws SQLException {
             batch(pos, null);
         }
         else if (value instanceof java.util.Date) {
-            setDateByMilliseconds(pos, ((java.util.Date) value).getTime());
-        }
-        else if (value instanceof Date) {
-            setDateByMilliseconds(pos, new Long(((Date) value).getTime()));
-        }
-        else if (value instanceof Time) {
-            setDateByMilliseconds(pos, new Long(((Time) value).getTime()));
-        }
-        else if (value instanceof Timestamp) {
-            setDateByMilliseconds(pos, new Long(((Timestamp) value).getTime()));
+            setDateByMilliseconds(pos, ((java.util.Date)value).getTime(), Calendar.getInstance());
```

Comment:
@xerial 
This is the main part of specification for setDate(int, Date, Calendar) method:

"Sets the designated parameter to the given java.sql.Date value,
using the given Calendar object.  The driver uses
the  Calendar object to construct an SQL DATE value,
which the driver then sends to the database.  With
a  Calendar object, the driver can calculate the date
taking into account a custom timezone.  If no
Calendar object is specified, the driver uses the default
timezone, which is that of the virtual machine running the application."

It says that driver can calculate the right date value using time zone from calendar. It also says that if there is no Calendar object, driver should use JVM time zone. So Calendar.getInstance() constructs calendar with JVM time zone. This is the way how other jdbc drivers works (for example postgresql jdbc driver and others) and i think sqlite jdbc should do the same things.

The problem is that old java time api is not fully time zone independent. All stuff are using default jvm time zone. It may be a problem, when, for example, your jdbc logic locates in different process which has another default time zone.

For example:
I construct date in client process which is correct and time zone independent (for example 2011-11-11 11:11:11 UTC). Then i send it to the jdbc process which has UTC+3 jvm time zone. Then i do setDate() (let clarify that date_format is text). SQLite formatting it as 2011-11-11 14:11:11 which is wrong. The purpose of Calendar object in jdbc API is to solve this problems, but currently sqlite jdbc just ignores Calendar in prepared statement (but doesn't in result set, which is inconsistent too).

Reply:
@xerial Any news?